spec/tasks/gen/model_spec.cr

```diff
@@ -0,0 +1,36 @@
+require "../../spec_helper"
+
+include CleanupHelper
+
+describe Gen::Model do
+  it "generates a model" do
+    with_cleanup do
+      io = IO::Memory.new
+      model_name = "ContactInfo"
+      ARGV.push(model_name)
+
+      Gen::Model.new.call(io)
+
+      File.read("./src/models/contact_info.cr").
+        should contain("class ContactInfo < BaseModel")
+      io.to_s.should contain("contact_info.cr")
+      io.to_s.should contain("./src/models")
+      File.read("./src/forms/contact_info_form.cr").
+        should contain("class ContactInfoForm < ContactInfo::BaseForm")
+      io.to_s.should contain("contact_info_form.cr")
+      io.to_s.should contain("./src/forms")
+      File.read("./src/queries/contact_info_query.cr").
+        should contain("class ContactInfoQuery < ContactInfo::BaseQuery")
+      io.to_s.should contain("contact_info_query.cr")
+      io.to_s.should contain("./src/queries")
+    end
+  end
+
+  it "displays an error if given no arguments" do
+    io = IO::Memory.new
+
+    Gen::Model.new.call(io)
+
+    io.to_s.should contain("Model name is required.")
+  end
+end
```

Comment:
What do you think about adding an error and a suggestion if they don’t have a camelcased name?

https://github.com/luckyframework/lucky_migrator/commit/722fb00b44f6490e94c0d37be6cfff48b41374da

Reply:
Good idea.  I'll add a regex matcher that ensures a capital letter and the absence of dashes and underscores.

Reply:
You could do what I did in migrator as well. Might be simpler/more accurate `model.camel_case != model`

Reply:
👍 .  I am continually impressed by what Crystal has in the standard library.

Reply:
Me too! It's a nice language :)